app/scripts/metamask-controller.js

```diff
@@ -568,12 +568,8 @@ export default class MetamaskController extends EventEmitter {
         ),
       getCurrentAccountEIP1559Compatibility:
         this.getCurrentAccountEIP1559Compatibility.bind(this),
-      legacyAPIEndpoint: `${gasApiBaseUrl}/networks/<chain_id>/gasPrices`,
       EIP1559APIEndpoint: `${gasApiBaseUrl}/networks/<chain_id>/suggestedGasFees`,
-      getCurrentNetworkLegacyGasAPICompatibility: () => {
-        const { chainId } = this.networkController.state.providerConfig;
-        return process.env.IN_TEST || chainId === CHAIN_IDS.MAINNET;
```

Comment:
I've since discovered that this legacy gas endpoint remains useful for non-EIP1559 chains that our APIs support (e.g. BSC and Polygon). But we can restore this for those chains in a later PR, they weren't using it yet anyway.